pygfx/renderers/wgpu/_utils.py

```diff
@@ -195,30 +197,99 @@ def generate_uniform_struct(dtype_struct, structname):
     return code
 
 
-class GpuCache:
-    """A chache for GPU objects."""
+class JsonEncoderWithWgpuSupport(json.JSONEncoder):
+    def default(self, ob):
+        if isinstance(ob, wgpu.GPUObjectBase):
+            return ob.__class__.__name__ + "@" + hex(id(ob))
+
+
+jsonencoder = JsonEncoderWithWgpuSupport()
+
+
+def hash_from_value(value):
+    """Simple way to create a hash from a (possibly composite) object.
+    Assumes JSON encodable objects and GPU objects.
+    """
+    # Encode the value to string using json. The JSON encoder is so fast that
+    # its hard to come up with something that can serialze to str faster.
+    s = jsonencoder.encode(value)
+
+    # If the strings is small enough, just use that. Otherwise use fast
+    # sha1 hashing to produce a 40-char hash. This costs a bit more
+    # time, but safes memory. Can also set this to True for debugging purposes.
+    if len(s) < 128:
+        return s
+    else:
+        h = hashlib.sha1()
+        h.update(s.encode())
+        return h.hexdigest()
```

Comment:
Would standard `hash()` be faster and simpler here?

Reply:
`hash()` is faster, but only by a bit. Sha1 is more secure and maps to a larger space (40 chars representing 160 bits, rather than an int representing 64 bits). Also because we assume the hash is a string, we can just use the json string when its short, saving the hash step altogether.

Reply:
Ok. I don't think security is a concern here, and I also don't think we will have to worry about hash collisions as the number of objects is not going into the millions.

Reply:
That's fair. I'll update to use `hash()` instead.